pkg/maps/ipcache/ipcache.go

```diff
@@ -159,18 +169,33 @@ func NewMap(name string) *Map {
 				return &k, &v, nil
 			},
 		).WithCache(),
+		deleteSupport: true,
 	}
 }
 
-// Delete removes a key from the ipcache BPF map
-func (m *Map) Delete(k bpf.MapKey) error {
+// delete removes a key from the ipcache BPF map, and returns whether the
+// kernel supports the delete operation (true) or not (false), and any error
+// that may have occurred while attempting to delete the entry.
+//
+// If "overwrite" is true, then if delete is not supported the entry's value
+// will be overwritten with zeroes to signify that it's an invalid entry.
+func (m *Map) delete(k bpf.MapKey, overwrite bool) (bool, error) {
 	// Older kernels do not support deletion of LPM map entries so zero out
 	// the entry instead of attempting a deletion
 	err, errno := m.DeleteWithErrno(k)
 	if errno == unix.ENOSYS {
-		return m.Update(k, &RemoteEndpointInfo{})
+		if overwrite {
+			return false, m.Update(k, &RemoteEndpointInfo{})
+		}
+		return false, nil
```

Comment:
technically this should be `false, err` but it doesn't matter for existing users.